contentcuration/contentcuration/static/js/edit_channel/import/views/ImportDialogue.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+
+  <div>
+    <!-- SEARCH FORM -->
+    <div>
+      <form @submit="submitSearch" :disabled="!searchTermIsValid" class="SearchForm">
+        <input
+          class="search-input"
+          v-model="searchTerm"
+          type="text"
+          placeholder="What are you looking for?"
+        />
+        <button
+          type="submit"
+          class="action-button modal-main-action-button"
+          @click.prevent="submitSearch"
+          :disabled="!searchTermIsValid"
+        >
+          Search
+        </button>
+      </form>
+    </div>
+
+    <!-- SLOT FOR TREE VIEW OR SEARCH RESULTS -->
+    <div id="import_from_channel_box" class="modal-content-default">
+      <slot />
+    </div>
+
+    <br/>
+
+    <div id="import_bottom_container" class="modal-bottom-content-default">
+      <a class="action-text" data-dismiss="modal">
+        <span>CANCEL</span>
+      </a>
+      <button
+        class="action-button pull-right modal-main-action-button"
+        id="import_content_submit"
+        @click="handleClickImport"
+        :disabled="!importIsEnabled"
+      >
+        <span v-if="!importIsEnabled">
+          Select content to import...
+        </span>
+        <span v-else>
+          IMPORT
+        </span>
+      </button>
+      <span id="import_file_metadata" class="pull-right">
+        <span id="import_file_count">
+          {{ topicCount | pluralize('Topic') }} {{ resourceCount | pluralize('Resource') }}
+        </span>
+        <em id="import_file_size">
+          {{ importFileSizeInWords | parenthesize }}
+        </em>
+      </span>
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+const stringHelper = require('../../utils/string_helper');
+const { hasRelatedContent } = require('../util');
+const { mapGetters, mapState, mapActions, mapMutations } = require('vuex');
+const  { pluralize, parenthesize } = require('./filters');
+
+module.exports = {
+  components: {
+    ImportChannelList: require('./ImportChannelList.vue'),
+  },
+  data() {
+    return {
+      searchTerm: '',
+    };
+  },
+  computed: {
+    ...mapState('import', [
+      'itemsToImport',
+      'importSizeInBytes',
+    ]),
+    ...mapGetters('import', [
+      'importedItemCounts',
+      'currentSearchTerm',
+      'currentImportPage',
+    ]),
+    searchTermIsValid() {
+      return this.searchTerm.length >= 3;
```

Comment:
To me a single character search seems like a user error and probably won't have performance issues, just provide useless results. I've set the API to return nothing if somehow a person requests "" as a search term because that query would technically return every item as a result.